tools/autograd/gen_variable_type.py

```diff
@@ -485,11 +485,32 @@ def is_differentiable(arg):
         if 'Tensor' not in arg['type']:
             return False
         if arg['dynamic_type'] in {'IndexTensor', 'BoolTensor'}:
+            # TODO: Enable this after native_functions.yaml schema unification.
+            # These are necessary for legacy code and should be
+            # used by legacy code only!
+            # assert name.startswith('_th_'), \
+            # "IndexTensor and BoolTensor are restricted to legacy _th_ functions only.
             return False
         return True
 
+    def find_args_with_derivatives(differentiable_inputs):
+        """Find arguments that have derivative definitions"""
+        if func is None:
+            return differentiable_inputs
+        names = set(name for d in func['derivatives'] for name in d['var_names'])
+        differentiable = [arg for arg in differentiable_inputs if arg['name'] in names]
+        if len(differentiable) != len(names):
+            missing = names - set(arg['name'] for arg in differentiable)
+            raise RuntimeError('Missing arguments for derivatives: {} in {}'.format(missing, func['name']))
+        return differentiable
+
     inputs = [arg for arg in arguments if not arg.get('output', False)]
     differentiable_inputs = list(filter(is_differentiable, inputs))
+    args_with_derivatives = find_args_with_derivatives(differentiable_inputs)
+    not_args_with_derivatives = []
+    if func:
+        for not_derivative in func['not_derivatives']:
+            not_args_with_derivatives += not_derivative['var_names']
```

Comment:
append?

Reply:
No, they need to be flattened. A formula can refer to multiple args.

Reply:
ah, my bad.  There are no cases where this is actually used for not_derivatives right?